lib/reverseproxy/revers_proxy.go

```diff
@@ -0,0 +1,126 @@
+package reverseproxy
+
+import (
+	"flag"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+const (
+	targetURL     = "vm-target-url"
+	forwardHeader = "X-Forwarded-For"
+)
+
+var (
+	maxIdleConnsPerBackend = flag.Int("maxIdleConnsPerBackend", 100, "The maximum number of idle connections vmauth can open per each backend host")
+)
+
+// ReversProxy represents simple revers proxy based on http.Client
+type ReversProxy struct {
+	client *http.Client
+}
+
+// New initialize revers proxy
+func New() *ReversProxy {
+	return &ReversProxy{
+		client: &http.Client{},
+	}
+}
+
+// ServeHTTP serve http requests
+func (rr *ReversProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	rr.handle(w, r)
+}
+
+// handle works with income request, update it to outcome and copy response to the requester
+func (rr *ReversProxy) handle(w http.ResponseWriter, r *http.Request) {
+	tURL, err := getTargetURL(r)
+	if err != nil {
+		logger.Panicf("BUG: unexpected error when parsing targetURL=%q: %s", targetURL, err)
+	}
+
+	rr.client.Transport = prepareTransport(tURL)
```

Comment:
modification for transport is not needed. Just modify proxyReq.url

Reply:
fixed